From f65e507bf2784de8b6ab96194b9320c3de8ca53a Mon Sep 17 00:00:00 2001
From: Alexey Minnekhanov <alexeymin@postmarketos.org>
Date: Sun, 27 Apr 2025 17:58:13 +0300
Subject: [PATCH] drm/msm/a5xx: Specify speedbins for A508/509/512

Specify speedbins for Adreno 508, 509 and 512 GPUs found in
SDM630/636/660 SoCs and their respective bits used for bitmasks
in opp-supported-hw setting, using ADRENO_SPEEDBINS macro, same
way as a6xx_catalog does it.

This will allow us to reuse common code for a5xx/a6xx to configure
opp tables and its supported-hw values.

Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/f65e507bf2784de8b6ab96194b9320c3de8ca53a
Scope: SDM630/660-common
Porting notes: Original A508/A509/A512 speedbin catalog applies unchanged.
---
diff --git a/drivers/gpu/drm/msm/adreno/a5xx_catalog.c b/drivers/gpu/drm/msm/adreno/a5xx_catalog.c
index babd320f3b736f..8086b1937c17b9 100644
--- a/drivers/gpu/drm/msm/adreno/a5xx_catalog.c
+++ b/drivers/gpu/drm/msm/adreno/a5xx_catalog.c
@@ -58,6 +58,12 @@ static const struct adreno_info a5xx_gpus[] = {
 		.quirks = ADRENO_QUIRK_LMLOADKILL_DISABLE,
 		.funcs = &a5xx_gpu_funcs,
 		.zapfw = "a508_zap.mdt",
+		.speedbins = ADRENO_SPEEDBINS(
+			{  0, 3}, /* default bin, same as 162 one */
+			{135, 1}, /* up to 647 MHz */
+			{146, 2}, /* up to 700 MHz */
+			{162, 3}, /* up to 775 MHz */
+		),
 	}, {
 		.chip_ids = ADRENO_CHIP_IDS(0x05000900),
 		.family = ADRENO_5XX,
@@ -76,6 +82,15 @@ static const struct adreno_info a5xx_gpus[] = {
 		.funcs = &a5xx_gpu_funcs,
 		/* Adreno 509 uses the same ZAP as 512 */
 		.zapfw = "a512_zap.mdt",
+		.speedbins = ADRENO_SPEEDBINS(
+			{  0, 6}, /* default bin, same as 157 one */
+			{ 78, 1}, /* up to 370 MHz */
+			{ 90, 2}, /* up to 430 MHz */
+			{122, 3}, /* up to 585 MHz */
+			{135, 4}, /* up to 647 MHz */
+			{146, 5}, /* up to 700 MHz */
+			{157, 6}, /* up to 750 MHz */
+		),
 	}, {
 		.chip_ids = ADRENO_CHIP_IDS(0x05010000),
 		.family = ADRENO_5XX,
@@ -108,6 +123,15 @@ static const struct adreno_info a5xx_gpus[] = {
 		.quirks = ADRENO_QUIRK_LMLOADKILL_DISABLE,
 		.funcs = &a5xx_gpu_funcs,
 		.zapfw = "a512_zap.mdt",
+		.speedbins = ADRENO_SPEEDBINS(
+			{  0, 6}, /* default bin, same as 157 one */
+			{ 78, 1}, /* up to 370 MHz */
+			{ 90, 2}, /* up to 430 MHz */
+			{122, 3}, /* up to 585 MHz */
+			{135, 4}, /* up to 647 MHz */
+			{146, 5}, /* up to 700 MHz */
+			{157, 6}, /* up to 750 MHz */
+		),
 	}, {
 		.chip_ids = ADRENO_CHIP_IDS(
 			0x05030002,

